src/encord_active/lib/common/filtering.py

```diff
@@ -33,15 +38,28 @@ def __hash__(self):
         return hash(json.dumps(self.dict(exclude_defaults=True)))
 
 
-def apply_filters(df: pd.DataFrame, filters: Filters):
+def apply_filters(df: pd.DataFrame, filters: Filters, pfs: ProjectFileStructure):
     filtered = df.copy()
     filtered["data_row_id"] = filtered.index.str.split("_", n=3).str[0:3].str.join("_")
     filtered["is_label_metric"] = filtered.index.str.split("_", n=3).str.len() > 3
 
+    # Add workflow stage column in workflow projects
```

Comment:
[curious]🤔 

Correct me if I'm wrong, but it seems that we are adding the `workflow_stage` column solely for the purpose of filtering, right?
If that is the case, I think we can move this part of the logic into `filter_workflow_stages` and just not add the column.

Reply:
Sure, can do, either way we are already putting this piece of content in the wrong place as the filters should only filter 😄.

Reply:
with this change they are more clustered so easier to change, but not great nevertheless